### PR TITLE
feat(FX-3170): refresh search results page

### PR DESCRIPTION
**Pull request by tr-ann**
Jira: [FX-3170](https://artsyproduct.atlassian.net/browse/FX-3170)

### Description
The purpose of this ticket is to refresh the styles and layout of the search results page in Force to match the latest design system.
Figma spec is [here](https://www.figma.com/file/0svJJjlcrCFBkVYzagKP3Z/FX-v3-Surfaces?node-id=4021%3A2880) 

### Changes
- Updated components style
- Updated tests for search app
- Removed unused files
- Small refactor of components

### Demo

[review app](https://pr-8882.artsy.net/search?term=andy)

### CODE REVIEW

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (dimatretyak) [inline src/v2/Apps/Search/Components/NavigationTabs.tsx:49]: **nitpick**: can we specify `10px` here right away?

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (dimatretyak) [inline src/v2/Apps/Search/Components/NavigationTabs.tsx:70]: **suggestion**: please clarify with Jonathan whether there should be such a style with the hover effect?

https://user-images.githubusercontent.com/3513494/132865254-71e83241-e774-4b25-8f90-4fbd9aa7e174.mp4

**Comment by dimatretyak** [comment]
**nitpick**:  pills is different from the design in Figma on small devices

Search Results for 'aa' | Artsy 2021-09-10 17-06-17

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (dzucconi) [reply src/v2/Apps/Search/Components/NavigationTabs.tsx:49]: These should just be `Pill` components

Comment (dimatretyak) [reply src/v2/Apps/Search/Components/NavigationTabs.tsx:49]: Oh, I didn't know that palette has [such a component](https://palette.artsy.net/elements/pill/#gatsby-focus-wrapper). Thanks!

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (dzucconi) [reply src/v2/Apps/Search/Components/NavigationTabs.tsx:69]: We have a component for doing the sizing correctly called `Sup`: https://palette-storybook.artsy.net/?path=/story/components-sup--default

File: src/v2/Apps/Search/Components/SearchMeta.tsx
Comment (dzucconi) [inline src/v2/Apps/Search/Components/SearchMeta.tsx:19]: You can use https://github.com/artsy/force/blob/master/src/v2/Components/MetaTags.tsx here

File: src/v2/Apps/Search/Components/GenericSearchResultItem.tsx
Comment (dzucconi) [inline src/v2/Apps/Search/Components/GenericSearchResultItem.tsx:67]: This component has a `lineClamp` prop — so you can get rid of that truncator below

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (dzucconi) [inline src/v2/Apps/Search/Components/NavigationTabs.tsx:162]: I'm sure this is fine; though generally we use this for things of a consistent width. It might be more appropriate to just use `BaseTabs` here — which is a free scroll with no snapping but has the gradient fade overlay as an affordance when items go off-screen. I _think_ it should work fine. I may be mistaken.

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (dzucconi) [inline src/v2/Apps/Search/Components/NavigationTabs.tsx:202]: `non-blocking:` if you see these `get` calls you can just replace them with the safe operator

```suggestion
    let count = aggregationFor(props, key)?.count ?? 0
```

**Review by dzucconi — APPROVED** [review]
Looks good. I think if you rebase with master you should get that acceptance test passing. I'd consider swapping in `BaseTabs` (we should maybe rename that) for `Swiper` to get the gradient affordance; but non-blocking.

File: src/v2/Apps/Search/Components/NavigationTabs.tsx
Comment (tr-ann) [reply src/v2/Apps/Search/Components/NavigationTabs.tsx:162]: Thanks Damon, but when I use BaseTabs I get a separator at the bottom of tabs
image

And I don't understand how I can remove it. Can you help me with it?

Comment (dzucconi) [reply src/v2/Apps/Search/Components/NavigationTabs.tsx:162]: Ah, OK maybe punt on this for now and I'll extract the underlying primitive so that we can follow up. Thanks for looking!

**Review by mdole — COMMENTED** [review]
Really nice work @tr-ann!! Just checked this PR out locally and there are a few small visual tweaks that we should make before merging. Also two questions I'll ask Jonathan in Slack: should these pills be sticky? And should a scroll rail be visible on mobile/small window width?

### Pills missing margin between: 
Screen Shot 2021-09-15 at 11 08 12 AM

---

### Filter button doesn't match design specs
It's a black/blue pill here instead of the minimal text + icon

Screen Shot 2021-09-15 at 11 09 06 AM

---

### Missing 1 space between text + numbers in pill

Design:
Screen Shot 2021-09-15 at 11 15 17 AM

Current:
Screen Shot 2021-09-15 at 11 13 49 AM

---

### Pill "selected" state differs from design:

Frankly, this might be fine or it might be something we should change everywhere at the same time. I'm not 100% sure if this is already being discussed with the design team.

Design: 
Screen Shot 2021-09-15 at 11 16 46 AM

Current:
Screen Shot 2021-09-15 at 11 12 07 AM

**Comment by mdole** [comment]
Jonathan got back to us on Slack - we don't want a scroll rail, we do want sticky.

**Comment by tr-ann** [comment]
@mdole thanks for the clarifying. I updated the pills style and made them sticky.
What about visible scroll bar, I think we need to investigate it separately, because the scroll disappear when scrolling at least one and (example in attached video)

https://user-images.githubusercontent.com/56556580/133429808-cb2ff9c0-6e51-42f4-861d-e1370349e20b.mov

I think a new style of filter button is common (not only for search results page), so it will be better to refresh it separately (maybe we need to create ticket for that?). What do you think?

And the last, pill selected state. We need to create another one variant for search tabs (now we have pill styles for filters, artist and text). There is discussion about it on [slack](https://artsy.slack.com/archives/C9XJKPY9W/p1631283326018400)

**Comment by dzucconi** [comment]
@mdole Regarding the Pill design discrepancies: this was already discussed: there's a Pill filter variant. That's what this should use. The design _is_ updated but it's not reflected in Palette yet. Leave these alone and DSWG will take care of the Pill updates down the line.

**Comment by dzucconi** [comment]
But the various spacing issues you point out are valid

**Comment by dzucconi** [comment]
I'll also pull out a new component for doing that scroll thing. I'll see if I can get to that today.

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

**Comment by mdole** [comment]
Thank you both for clarifying! @tr-ann I think we can use the HorizontalOverflow component Damon added to Palette in [this PR](https://github.com/artsy/palette/pull/1034) to address the scrollbar behavior, and then I think we're ready to merge!

Because this is a large visual change, I would also suggest that we notify Jonathan and Thomas and make sure they have time to QA shortly after it is deployed to staging. Or, as an alternative, we could spin up a review app (see instructions [here](https://github.com/artsy/force/blob/bd325f12d3f8296cabdfab6901f9edbdad9ab9d9/docs/creating_review_app.md)) to give them time to QA without blocking deployments to production.

@tr-ann want to try creating a review app first, and if it's not too much work, we can use that for QA before we merge this PR? Please let me know if I can help!

**Comment by tr-ann** [comment]
Sure, thanks @mdole. I'll use HorisontalComponent and create review app after that

**Comment by tr-ann** [comment]
@mdole I notice interesting thing. BaseTabs component implementation already include HorizontalOverflow component ([here](https://github.com/artsy/palette/blob/694063c1a8ec3e42d36014e81541b08879cd6217/packages/palette/src/elements/BaseTabs/BaseTabs.tsx#L34)), but I don't understand why scroll bar displayed anyway.
@dzucconi can you clarify it?

**Comment by dzucconi** [comment]
@tr-ann I extracted that `HorizontalOverflow` so that you didn't have to use the `BaseTabs` (which has a border and in-built spacing). Though I guess this is fine.

WRT to the scrollbar. This is what it's using: https://github.com/artsy/palette/blob/ea901c51f6fbbed8a89c6aa87e973dd10c306325/packages/palette/src/helpers/visuallyDisableScrollbar.ts#L6-L15 — it's possible that there's just some browser setting there that's overruling it. Feel free to PR this mixin in Palette since you have a repro; but I'd punt as I suspect it's not affecting a major percentage. Also it's not like it's a bad thing that it's more usable IMO 😬

**Comment by mdole** [comment]
While looking at the review app, I noticed something that is also an issue with the current search results page (it is NOT new with this PR): when you click an artwork, your search term is replaced by “Undefined”. See the ticket below for gifs.

I’ve ticketed that [here](https://artsyproduct.atlassian.net/browse/FX-3328), just mentioning in this ticket in case anyone else notices it. Considering this is not new behavior, I would suggest we don’t block this PR and instead follow up!

**Comment by dzucconi** [comment]
Yeah that seach bug is very long standing and I believe ticketed within FX?

**Pull request merged**
